aspnet/mvc/controllers/testing.rst

```diff
@@ -1,8 +1,158 @@
-.. include:: /../common/stub-topic.txt
+Testing Controller Logic
+========================
+By `Steve Smith`_
 
-|stub-icon| Testing Controller Logic
-====================================
+Controllers in ASP.NET MVC apps should be small and focused on user-interface concerns. Provided this recommendation is followed, testing your app's controller logic should be fairly straightforward.
 
-.. include:: /../common/stub-notice.txt
+.. contents:: Sections
+	:local:
+	:depth: 1
+	
+`View sample files <https://github.com/aspnet/Docs/tree/1.0.0-rc1/aspnet/mvc/controllers/testing/sample>`_
 
-.. _issue: https://github.com/aspnet/Docs/issues/123
+What is Controller Logic
+------------------------
+*Controllers* define groups of *actions*. The framework could have been designed around isolated actions, but grouping them using controllers often useful for :doc:`routing </fundamentals/routing>`, applying :doc:`filters <filters>`, and :doc:`injecting common dependencies <dependency-injection>`. 
```

Comment:
How about something like:

*Controllers* define groups of [related] *actions*. The grouping of related actions into controllers is useful for